src/main.rs

```diff
@@ -101,34 +108,46 @@ fn main() -> ! {
     info!("Startup tc2-firmware {}", FIRMWARE_VERSION);
 
     // TODO: Check wake_en and sleep_en registers to make sure we're not enabling any clocks we don't need.
-    let mut peripherals = Peripherals::take().unwrap();
+    let mut peripherals: Peripherals = Peripherals::take().unwrap();
+    let is_audio = read_is_audio_from_rp2040_flash();
+
+    let freq;
+
+    if is_audio {
```

Comment:
I think this is an okay solution for now, but we'll need to allocate time to get to the bottom of the clock speed differences, since this also means that any timers we've tuned that are in shared code paths – like the one for the pi ping timeout – will be wrong for one or the other.

Very small style nit - usually I'd set something like the `freq` variable like `let freq = if is_audio { ROSC_TARGET_CLOCK_FREQ_HZ_AUDIO.Hz() } else { ROSC_TARGET_CLOCK_FREQ_HZ_THERMAL.Hz() };`